File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/data/DataControl.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Control import Control

@PythonTaskInfo
class DataControl(PythonTask):

    def __init_ports__(self):
        self.in_ = InputPort(name="in", data_type=Control)
        self.out = OutputPort(name="out", data_type=Control)

    def execute(self, grp):
        self.in_.get_data(Control);
        self.out.put_data(Control());

File: com.systemincloud.examples.tasks.rtask/src/test/r/data/DataControl.R

library(R6)
library(sicrtask)

DataControl <- R6Class("DataControl",
  inherit = RTask,
  public = list(
    rtaskinfo = RTaskInfo$new(),
    in_ = NA,
    out = NA,
    initialize_ports = function() {
      self$in_ <- InputPort$new(name = "in")
      self$out <- OutputPort$new(name = "out")
    },
    initialize = function() {
      self$initialize_ports()
    },
    execute = function(grp) {
	  self$in_$get_data("Control");
	  self$out$put_data(Control$new());
	}
  ),
  private = list(
  )
)
